pyuvdata/uvdata/uvdata.py

```diff
@@ -5563,14 +5570,179 @@ def compress_by_redundancy(self, tol=1.0, inplace=True, keep_all_metadata=True):
+                if len(conjugates) > 0:
+                    conj_group = set(group).intersection(conjugates)
+                    reg_group = list(set(group) - conj_group)
+                    conj_group = list(conj_group)
+                else:
+                    reg_group = group
+                    conj_group = []
+                group_times = []
+                group_inds = []
+                conj_group_inds = []
+                conj_group_times = []
+                for bl in reg_group:
+                    bl_inds = np.where(self.baseline_array == bl)[0]
+                    group_inds.extend(bl_inds)
+                    group_times.extend(self.time_array[bl_inds])
+                for bl in conj_group:
+                    bl_inds = np.where(self.baseline_array == bl)[0]
+                    conj_group_inds.extend(bl_inds)
+                    conj_group_times.extend(self.time_array[bl_inds])
+
+                group_inds = np.array(group_inds, dtype=np.int)
+                conj_group_inds = np.array(conj_group_inds, dtype=np.int)
+                # now we have to figure out which times are the same to a tolerance
+                # so we can average over them.
+                time_inds = np.arange(len(group_times + conj_group_times))
+                time_gps = uvutils.find_clusters(
+                    time_inds,
+                    np.array(group_times + conj_group_times),
+                    self._time_array.tols[1],
+                )
+
+                # average over the same times
+                obj_bl = bl_ants[grp_ind]
+                obj_inds = new_obj._key2inds(obj_bl)[0]
+                obj_times = new_obj.time_array[obj_inds]
+
+                for gp in time_gps:
+                    avg_time = np.average(np.array(group_times + conj_group_times)[gp])
+
+                    obj_time_ind = np.where(
+                        np.abs(obj_times - avg_time) < self._time_array.tols[1]
+                    )[0]
+
+                    if obj_time_ind.size == 1:
+                        this_obj_ind = obj_inds[obj_time_ind[0]]
+                        temp_time_array[this_obj_ind] = avg_time
+                    else:
+                        warnings.warn(
+                            "Index baseline in the redundant group does not "
+                            "have all the times, compressed object will be "
+                            "missing those times."
+                        )
+                        continue
+
+                    regular_orientation = np.array(
+                        [time_ind for time_ind in gp if time_ind < len(group_times)],
+                        dtype=np.int,
+                    )
+                    regular_inds = group_inds[np.array(regular_orientation)]
+                    conj_orientation = np.array(
+                        [time_ind for time_ind in gp if time_ind >= len(group_times)],
+                        dtype=np.int,
+                    )
```

Comment:
The ` < len(group_times)` took me  second to understand, since that's the barrier between normally and conjugated baselines in the array, maybe we can add a inline comment to that effect?

Reply:
Yes, I'll add a comment.